devicetypes/smartthings/fidure-thermostat.src/fidure-thermostat.groovy

```diff
@@ -292,7 +292,7 @@ def parseDescriptionAsMap(description) {
     def result = map.get('result')
     def list = [];
 
-  if (getDataLengthByType(map.get('encoding')) < map.get('value').length()) {
+  if (getDataLengthByType(map.get('encoding')) != null && getDataLengthByType(map.get('encoding')) < map.get('value').length()) {
```

Comment:
hmmmmm Is there the chance that `map.get('encoding')` can be null? In which case getDataLengthByType will have an issue when it goes for `map.get(t)` (map.get(null)). I almost wonder if it is possible to just dump this function altogether and use the one in the Zigbee library? I'm not quite sure how much custom stuff is in here. I see a few other uses without null protection, like `map.get('value').length()` that has me concerned...

Now that I am looking at the files as a whole, I almost want to re-write the entire thing.....

Reply:
It might be that `map.get('encoding')` is `null`. I'm mostly just doing naive-as-possible fixes since I don't want to touch that DTH any more than I have to. It's very nasty.